leases/schema/mutations.py

```diff
@@ -189,6 +189,69 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteBerthLeaseMutation()
 
 
+class SwitchBerthMutation(graphene.ClientIDMutation):
+    class Input:
+        old_lease_id = graphene.ID(required=True)
+        new_berth_id = graphene.ID(required=True)
+        switch_date = graphene.Date(
+            description="The date which will mark the end of the old lease and the start of the new lease."
+            "If none is provided, it will default to the day when the mutation is called"
+        )
+
+    old_berth_lease = graphene.Field(BerthLeaseNode)
+    new_berth_lease = graphene.Field(BerthLeaseNode)
+
+    @classmethod
+    @add_permission_required(BerthLease)
+    @change_permission_required(BerthLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **input):
+        old_lease_id = input.pop("old_lease_id")
+        new_berth_id = input.pop("new_berth_id")
+        switch_date = input.get("switch_date") or calculate_berth_lease_start_date()
+
+        old_lease = get_node_from_global_id(
+            info, old_lease_id, only_type=BerthLeaseNode, nullable=False,
+        )
+
+        try:
+            contract = old_lease.contract
+        except ObjectDoesNotExist as e:
+            # Flatten all the error messages on a single list
+            raise VenepaikkaGraphQLError(str(e))
+
+        kwargs = {
+            "boat": old_lease.boat,
+            "customer": old_lease.customer,
+            "end_date": old_lease.end_date,
+            "renew_automatically": old_lease.renew_automatically,
+            "status": old_lease.status,
+        }
+
+        old_lease = terminate_lease(
+            info, old_lease_id, BerthLeaseNode, end_date=switch_date,
+        )
+        old_lease.comment = "Lease is terminated due to berth switch"
+        old_lease.save(update_fields=["comment"])
+
+        new_berth = get_node_from_global_id(
+            info, new_berth_id, only_type=BerthNode, nullable=False,
+        )
+
+        try:
+            kwargs.update({"berth": new_berth, "start_date": switch_date})
```

Comment:
1. Can you rename `kwargs` to (e.g.) `lease_args` to make it a bit clearer?
2. Can you define the whole `lease_args` here instead?
3. **OR** can you pass the params directly to the `create` instead of first creating a dict and then passing it?

Reply:
1. Yep, let's rename.
2 / 3. I'm defining the kwargs before the termination to get the data of the old lease before the termination just in case. Now that I think about it, nothing is changed during the termination except for the status and comment fields so should be safe to pass them straight to the creation after the termination.